homeassistant/components/switch/wemo.py

```diff
@@ -110,11 +111,30 @@ def device_state_attributes(self):
         if self.insight_params or (self.coffeemaker_mode is not None):
             attr[ATTR_CURRENT_STATE_DETAIL] = self.detail_state
 
+        if self.insight_params:
+            attr['on_latest_time'] = \
+                WemoSwitch.as_uptime(self.insight_params['onfor'])
+            attr['on_today_time'] = \
+                WemoSwitch.as_uptime(self.insight_params['ontoday'])
+            attr['on_total_time'] = \
+                WemoSwitch.as_uptime(self.insight_params['ontotal'])
+            attr['power_threshold_mw'] = \
+                self.insight_params['powerthreshold']
+
         if self.coffeemaker_mode is not None:
             attr[ATTR_COFFEMAKER_MODE] = self.coffeemaker_mode
 
         return attr
 
+    @staticmethod
+    def as_uptime(_seconds):
+        """Format seconds into uptime string in the format: 00d 00h 00m 00s."""
+        uptime = datetime(1, 1, 1) + timedelta(seconds=_seconds)
+        return "{:0>2d}d {:0>2d}h {:0>2d}m {:0>2d}s".format(uptime.day-1,
```

Comment:
I'm probably doing something stupid (python noob), but when I try that I get this:
```
>>> from datetime import datetime, timedelta
>>> uptime = timedelta(days=0,seconds=86401,hours=0,minutes=0)
>>> print ("{:0>2d}d {:0>2d}h {:0>2d}m {:0>2d}s".format(uptime.days, uptime.hours, uptime.minutes, uptime.seconds))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'datetime.timedelta' object has no attribute 'hours'
```